packages/flutter/lib/src/cupertino/text_selection_toolbar_buttons_builder.dart

```diff
@@ -0,0 +1,113 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'debug.dart';
+import 'desktop_text_selection_toolbar_button.dart';
+import 'localizations.dart';
+import 'text_selection_toolbar_button.dart';
+
+/// A Function that builds a context menu given a List of its children Widgets.
+///
+/// See also:
+///
+///   * [CupertinoTextSelectionToolbarButtonsBuilder], which receives an
+///     instance of this as a parameter.
+typedef ContextMenuFromChildrenBuilder = Widget Function(
+  BuildContext context,
+  List<Widget> children,
+);
+
+/// Calls [builder] with a List of Widgets generated by turning [buttonItems]
+/// into the default Cupertino buttons for the platform.
+///
+/// Does not build Material buttons. On non-Apple platforms, Cupertino-style
+/// buttons will still be used, because the Cupertino library does not access
+/// the Material library.
+///
+/// See also:
+///
+/// * [DefaultTextSelectionToolbar], which builds the toolbar itself. By
+///   wrapping [TextSelectionToolbarButtonsBuilder] with
+///   [DefaultTextSelectionToolbar] and passing the given children to
+///   [DefaultTextSelectionToolbar.children], a default toolbar can be built
+///   with custom button actions and labels.
+/// * [TextSelectionToolbarButtonsBuilder], which is in the Material library and
+///   builds native-looking buttons for all platforms.
+/// * [EditableTextContextMenuButtonItemsBuilder], which is similar to this class,
+///   but calls its builder with [ContextMenuButtonItem]s instead of with fully
+///   built children Widgets.
+class CupertinoTextSelectionToolbarButtonsBuilder extends StatelessWidget {
+  /// Creates an instance of [CupertinoTextSelectionToolbarButtonsBuilder].
+  const CupertinoTextSelectionToolbarButtonsBuilder({
+    super.key,
+    required this.buttonItems,
+    required this.builder,
+  });
+
+  /// The information used to create each button Widget.
+  final List<ContextMenuButtonItem> buttonItems;
+
+  /// Called with a List of Widgets created from the given [buttonItems].
+  ///
+  /// Typically builds a text selection toolbar with the given Widgets as
```

Comment:
```suggestion
  /// Typically builds a text selection toolbar with the Widgets it is given as
```
(the widgets aren't given here, they're given when the builder is called)